src/applications/claims-status/actions/index.jsx

```diff
@@ -147,16 +148,51 @@ export function fetchClaimsSuccess(response) {
   };
 }
 
-export function getClaimsV2() {
+const POLLING_INTERVAL = 1000;
+
+export function pollClaimsStatus({
+  onFailure,
+  onSuccess,
+  pollingInterval,
+  request = apiRequest,
+  shouldFail,
+  shouldSucceed
+}) {
+  return request(
```

Comment:
Not really sure what the best composition is. The polling logic is just `setTimeout` with a recursive call but I was thinking it would be good to have the forking behavior grouped i.e. 
- does it poll again?
- does it succeed?
- does it fail? 

I could pass another predicate / callback pair in like `shouldPoll` and `onPoll` so it's more consistent. Then I wouldn't need to pass in the polling interval. 

![screen shot 2018-08-17 at 12 11 23](https://user-images.githubusercontent.com/4998130/44281959-b1287900-a216-11e8-933f-18446e115ca1.png)

Reply:
Okay I see what you mean- I'll refactor this a bit.

Reply:
Are you talking about drying the `get`? 

```
shouldFail: response => _.get(response, 'meta.syncStatus') === 'FAILED',
shouldSucceed: response => _.get(response, 'meta.syncStatus') === 'SUCCESS'
```

Reply:
Precisely.

Reply:
Refactored that get to use the platform and added a helper for that.

I tried a couple of different ways to reorganize this but decided to leave it mostly the same. Couple differences from the burials polling example:
- It takes a timeout first approach which means the initial request doesn't fire until the first polling timeout 
- It's not using the `apiRequest` helper which has our error reporting baked in and is not promisified